src/renderer/rendered_window.rs

```diff
@@ -353,11 +354,44 @@ impl RenderedWindow {
             .any(|line| line.borrow().has_transparency)
     }
 
+    fn draw_window_background_image(
+        paint: &Paint,
+        canvas: &Canvas,
+        image: &Image,
+        window_rect: &Rect,
+        screen_rect: &Rect,
+    ) {
+        let image_width = image.width() as f32;
+        let image_height = image.height() as f32;
+
+        // Calculate the scaling factors based on the full screen size
+        let scale_x = image_width / screen_rect.width();
+        let scale_y = image_height / screen_rect.height();
+
+        // Calculate the portion of the image that corresponds to the window's position
+        let src_x = scale_x * (window_rect.left() - screen_rect.left());
+        let src_y = scale_y * (window_rect.top() - screen_rect.top());
+        let src_width = scale_x * window_rect.width();
+        let src_height = scale_y * window_rect.height();
+
+        let src_rect = Rect::from_xywh(src_x, src_y, src_width, src_height);
```

Comment:
Since this implies unequal scaling on both axes, could this be made a cropping operation, such that both axes are always scaled equally and the longest side gets pixels cut off?